core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java

```diff
@@ -73,7 +73,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         if ("_name".equals(currentFieldName)) {
                             queryName = parser.text();
                         } else if ("value".equals(currentFieldName) || "prefix".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.textOrNull();
```

Comment:
reasoning here is that we check for null at the end and return a probably more comprehensive error message.